tasks/fullStack/app/models/shop.rb

```diff
@@ -1,4 +1,27 @@
 class Shop < ApplicationRecord
   has_many :products
   has_many :reviews, through: :products
+  has_many :monthly_avg_ratings
+
+  def ratings_change(months)
+    today = Date.today
+    x_months_ago = today - months.months
+
+    avg_ratings = monthly_avg_ratings
+      .where(start_date: x_months_ago.end_of_month..today.end_of_month)
```

Comment:
I think with this condition, it will return data of the last 4 months (but not just the last 3 months), June, July, August and September. Not a big issue but not exactly as in the requirements I think.

Reply:
I think the result also includes reviews on the day `2024-06-30`? So there are actually 4 months, including June (although just 1 day of June)?

Reply:
Let me clarify a bit. The `start_date` in the database is always the first day of the month, meaning that if we query data between `2024-06-30` and `2024-09-01`, it would not return data for June. Because the `start_date` is set to the beginning of each month, June's data would only be included if the range started on or before `2024-06-01`.